client/app/scripts/components/node-details.js

```diff
@@ -77,14 +77,19 @@ const NodeDetails = React.createClass({
     const details = this.props.details;
     const nodeExists = this.props.nodes && this.props.nodes.has(this.props.nodeId);
 
+    if (details) {
+      return this.renderDetails();
+    }
+
```

Comment:
Sounds fine, was just checking that `renderDetails` wouldn't "panic" (js, lol), if `!nodeExists`, but it looks like it won't.